.github/workflows/actionTests.yaml

```
@@ -0,0 +1,55 @@
+name: EngageSphere tests 🧪
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the repository
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Setup Node version
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+
+      # Install and run backend
+      - name: Install and run backend
+        run: |
+          cd backend
+          npm install
+          npm start &
+        env:
+          CI: true
+
+      # Install and run frontend
+      - name: Install and run frontend
+        run: |
+          cd frontend
+          npm install
+          npm start &
+        env:
+          CI: true
+
+      # Install and run tests in the root folder
+      - name: Install and run tests
+        run: |
+          npm install
+          npm run test
```

Visto que o script `test` é um padrão do npm, aqui poderia ser simplesmente:

```suggestion
          npm test
```